tests/test_resolver.py

```diff
@@ -119,6 +119,30 @@ async def test_threaded_negative_lookup() -> None:
         await resolver.resolve("doesnotexist.bla")
 
 
+async def test_threaded_negative_lookup_with_unknown_result() -> None:
+    loop = Mock()
+
+    # If compile CPython with `--disable-ipv6` option,
+    # we will get an (int, bytes) tuple, instead of a Exception.
+    async def unknown_addrinfo(*args: Any, **kwargs: Any) -> List[Any]:
+        return [
+            (
+                socket.AF_INET6,
+                socket.SOCK_STREAM,
+                6,
+                "",
+                (10, b"\x01\xbb\x00\x00\x00\x00*\x04NB\x00\x1a\x00\x00"),
+            )
+        ]
+
+    loop.getaddrinfo = unknown_addrinfo
```

Comment:
I would switch to unittest.MagicMock in this case.
Something like
```
loop.getaddrinfo = MagicMock(return_value=[
            (
                socket.AF_INET6,
                socket.SOCK_STREAM,
                6,
                "",
                (10, b"\x01\xbb\x00\x00\x00\x00*\x04NB\x00\x1a\x00\x00"),
            )
        ])
```

Reply:
Need to use `AsyncMock` here, but it was added in Python 3.8. There are Python 3.7 tests on the current CI.